Aardvark/ARKEmailBugReporter.m

```diff
@@ -342,169 +336,143 @@ - (NSString *)formattedLogMessagesAttachmentExtension;
-    while (viewControllerToPresentAlertController.presentedViewController != nil) {
-        viewControllerToPresentAlertController = viewControllerToPresentAlertController.presentedViewController;
-    }
-
-    /*
-     Disabling animations here to avoid potential crashes resulting from unexpected view state in UIKit
-     */
-    [viewControllerToPresentAlertController presentViewController:alertController animated:NO completion:NULL];
-
 }
 
-- (void)_configureAlertTextfield:(UITextField *)textField
+- (ARKEmailBugReportConfiguration *)_configurationWithCurrentSettings;
 {
-    textField.autocapitalizationType = UITextAutocapitalizationTypeSentences;
-    textField.autocorrectionType = UITextAutocorrectionTypeYes;
-    textField.spellCheckingType = UITextSpellCheckingTypeYes;
-    textField.returnKeyType = UIReturnKeyDone;
-}
-
-- (void)_createBugReportWithTitle:(NSString *)title;
-{
-    NSArray *logStores;
+    ARKEmailBugReportConfiguration *const configuration = [ARKEmailBugReportConfiguration new];
+    
     if (self.emailAttachmentAdditionsDelegate != nil) {
         NSMutableArray *const filteredLogStores = [NSMutableArray arrayWithCapacity:self.logStores.count];
         for (ARKLogStore *logStore in self.logStores) {
             if ([self.emailAttachmentAdditionsDelegate emailBugReporter:self shouldIncludeLogStoreInBugReport:logStore]) {
                 [filteredLogStores addObject:logStore];
             }
         }
-        logStores = filteredLogStores;
+        configuration.logStores = filteredLogStores;
+        
+        configuration.additionalAttachments = [self.emailAttachmentAdditionsDelegate additionalEmailAttachmentsForEmailBugReporter:self];
+        
     } else {
-        logStores = [self.logStores copy];
+        configuration.logStores = [self.logStores copy];
     }
     
+    configuration.includesScreenshot = self.attachScreenshotToNextBugReport;
+    configuration.includesViewHierarchyDescription = (self.attachScreenshotToNextBugReport && self.attachesViewHierarchyDescriptionWithScreenshot);
+    
+    return configuration;
+}
+
+- (void)_createBugReportWithConfiguration:(ARKEmailBugReportConfiguration *)configuration;
+{
     NSMapTable *logStoresToLogMessagesMap = [NSMapTable new];
     NSDictionary *emailBodyAdditions = [self.emailBodyAdditionsDelegate emailBodyAdditionsForEmailBugReporter:self];
     
     if ([MFMailComposeViewController canSendMail]) {
         self.mailComposeViewController = [MFMailComposeViewController new];
         
         [self.mailComposeViewController setToRecipients:@[self.bugReportRecipientEmailAddress]];
-        [self.mailComposeViewController setSubject:title];
+        [self.mailComposeViewController setSubject:configuration.prefilledEmailSubject];
+        
+        dispatch_group_t logStoreRetrievalDispatchGroup = dispatch_group_create();
+        dispatch_group_enter(logStoreRetrievalDispatchGroup);
```

Comment:
It would be a little better to put this outside the `canSendMail` check so the bug is fixed for both code paths.